Bottom: e99be4fe31a6aab090e5cff2e33258e1b9e3c0a4
Top:    3901e8ecd7605c5aec71b1ebb5b7362358193923
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-08 09:31:52 +0800

Refresh of autofs-5.0.7-fix-potential-null-dereference-in-lookup_mount.patch

---

diff --git a/modules/lookup_file.c b/modules/lookup_file.c
index 48f5789..eb6e9dc 100644
--- a/modules/lookup_file.c
+++ b/modules/lookup_file.c
@@ -1130,19 +1130,7 @@ do_cache_lookup:
 	ret = ctxt->parse->parse_mount(ap, key, key_len,
 				       mapent, ctxt->parse->context);
 	if (ret) {
-		time_t now = time(NULL);
-		int rv = CHE_OK;
-
-		cache_writelock(mc);
-		me = cache_lookup_distinct(mc, key);
-		if (!me)
-			rv = cache_update(mc, source, key, NULL, now);
-		if (rv != CHE_FAIL) {
-			me = cache_lookup_distinct(mc, key);
-			if (me)
-				me->status = now + ap->negative_timeout;
-		}
-		cache_unlock(mc);
+		cache_update_negative(mc, source, key, ap->negative_timeout);
 		return NSS_STATUS_TRYAGAIN;
 	}
 
diff --git a/modules/lookup_ldap.c b/modules/lookup_ldap.c
index 92f955e..68aa450 100644
--- a/modules/lookup_ldap.c
+++ b/modules/lookup_ldap.c
@@ -3011,20 +3011,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 	ret = ctxt->parse->parse_mount(ap, key, key_len,
 				       mapent, ctxt->parse->context);
 	if (ret) {
-		time_t now = time(NULL);
-		int rv = CHE_OK;
-
-		/* Record the the mount fail in the cache */
-		cache_writelock(mc);
-		me = cache_lookup_distinct(mc, key);
-		if (!me)
-			rv = cache_update(mc, source, key, NULL, now);
-		if (rv != CHE_FAIL) {
-			me = cache_lookup_distinct(mc, key);
-			if (me)
-				me->status = now + ap->negative_timeout;
-		}
-		cache_unlock(mc);
+		cache_update_negative(mc, source, key, ap->negative_timeout);
 		return NSS_STATUS_TRYAGAIN;
 	}
